serde-reflection/src/format.rs

```diff
@@ -85,12 +99,16 @@ pub struct Named<T> {
     pub value: T,
 }
 
+#[derive(Debug, Clone, Default, Eq, PartialEq)]
+/// A mutable holder for an initially unknown value.
+pub struct Variable<T>(Rc<RefCell<Option<T>>>);
+
 #[derive(Serialize, Deserialize, Debug, Clone, Eq, PartialEq)]
 #[serde(rename_all = "UPPERCASE")]
 /// Description of a variant in an enum.
 pub enum VariantFormat {
-    /// A variant whose format is yet unknown.
-    Unknown,
+    /// (used internally for tracing) A variant whose format is initially unknown.
+    Variable(Variable<VariantFormat>),
```

Comment:
If this is for internal use only then you could always #[doc(hidden)] this

Reply:
I'm thinking a user may still wish to write `Variable(_) => unreachable!()` instead of `_ => unreachable!()` at the end of a pattern matching.

I also thought about having a separate public type but one would need to change trace_type and trace_format APIs (to not return a temporary format) and make unit tests more complicated.